src/core/biblio.js

```diff
@@ -93,24 +93,52 @@ export function wireReference(rawRef, target = "_blank") {
   `;
 }
 
-export function stringifyReference(ref) {
+// Author, A. (Year, Month Date of Publication). Article title. Retrieved from URL
+function entryToAPA(ref) {
+	const authors = ((ref.authors && ref.authors.length) ? (ref.authors.join("; ") + (ref.etAl ? " et al." : ".")) : "");
+	const date = ref.date ?  "(" + ref.date + "). " : "";
+	const title = ref.href ? `<a href="${ref.href}"> <cite>${ref.title}</cite> </a>. ` : "";
+	const URL = ref.href ? `Retrieved from URL: <a href="${ref.href}">${ref.href}</a>` : "";
+
+    return `${authors} ${date} ${title} ${URL}`;
+}
+
+//Author’s Last name, First name. “Title of the Article or Individual Page.” Title of the website, Name of the publisher, Date of publication, URL.
+function entryToMLA(ref) {
+	const authors = ((ref.authors && ref.authors.length) ? (ref.authors.join("; ") + (ref.etAl ? " et al." : ".")) : "");
+	const date = ref.date ?  ref.date + ", " : "";
+	const title = ref.href ? `<a href="${ref.href}"> <cite>${ref.title}</cite> </a>. ` : "";
+	const URL = ref.href ? `<a href="${ref.href}">${ref.href}</a>` : "";
+	const publisher = ref.publisher ? ref.publisher + (/\.$/.test(ref.publisher) ? "" : ",") : "";
+      	 
+  	return 	`${authors} "${title}" ${publisher} ${date} ${URL}`;
+}
+
+// what ReSpec currently does...
+function entryToW3C(ref) {
+	const title = ref.href ? `<a href="${ref.href}"> <cite>${ref.title}</cite> </a>. ` : "";
+	const authors = ((ref.authors && ref.authors.length) ? (ref.authors.join("; ") + (ref.etAl ? " et al." : ".")) : "");
+	const publisher = ref.publisher ? ref.publisher + (/\.$/.test(ref.publisher) ? "" : ".") : "";
+	const date = ref.date ?  ref.date + ". " : "";
+	const status = ref.status ? (REF_STATUSES.get(ref.status) || ref.status) + ". " : "";
+	const URL = ref.href ? `URL: <a href="${ref.href}">${ref.href}</a>` : "";
+
+	return `${title} ${authors} ${publisher} ${date} ${status} ${URL}`;
+}
+
+export function stringifyReference(ref,style) {
   if (typeof ref === "string") return ref;
-  let output = `<cite>${ref.title}</cite>`;
-  if (ref.href) {
-    output = `<a href="${ref.href}">${output}</a>. `;
-  }
-  if (ref.authors && ref.authors.length) {
-    output += ref.authors.join("; ");
-    if (ref.etAl) output += " et al";
-    output += ".";
-  }
-  if (ref.publisher) {
-    const publisher = ref.publisher + (/\.$/.test(ref.publisher) ? "" : ".");
-    output = `${output} ${publisher} `;
+  let output = "";
```

Comment:
Ok, I think what we want to do here is "normalize" each reference. That way, we don't need to check for missing properties, types, etc. 
 
```JS
// this would  go outside stringifyReference()
const canonicalRef = {
   authors: [],
   date: "",
   status: "",
   href: "",
   publisher: "",
};

// Then in stringifyReference(), we plug all the holes!
const nomarlizedRef = {...canonicalRef, ref};
```

Reply:
fixed typo above